storage_plugin/functions/dashboard/main.ts

```diff
@@ -0,0 +1,138 @@
+        <title>Document</title>
+        <script src="https://cdn.tailwindcss.com"></script>
+      </head>
+      <body class="py-5">
+        <div class="flex flex-col items-center gap-3 w-full">
+          <form action="${exampleBaseUrlWithObject}" method="post" enctype="multipart/form-data" class="flex flex-col items-center gap-2">
+            <label for="uploader">
+              <input id="uploader" name="file" type="file" class="hidden" />
+              <div class="border rounded-md p-2 w-max">chose file</div>
+            </label>
+            <button type="submit" class="border rounded-md p-2 w-max">创建资源</button>
+          </form>
+          <ul class="flex flex-col gap-2 w-1/2">
+            ${
+    objects.map(
+      (object) =>
+        html`
+                  <li class="flex items-center justify-between py-4 px-5 border rounded-md">
+                    <div class="flex flex-col gap-1">
+                      <p>ID: ${object.id}</p>
+                      <p>Key: ${object.key}</p>
+                    </div>
+                    <div class="flex gap-2 ml-auto">
+                      <form action="${exampleBaseUrlWithObject}/${object.id}" method="get">
+                        <button type="submit" class="cursor-pointer">show</button>
+                      </form>
+                      <form action="${exampleBaseUrlWithObject}/${object.id}" method="post">
+                        <input type="text" name="_method" value="DELETE" class="hidden" />
+                        <button type="submit" class="cursor-pointer">delete</button>
+                      </form>
+                    </div>
+                  </li>
+                `,
+    )
+  }
+          </ul>
+        </div>
+      </body>
+    </html>
+  `);
+});
+
+app.post("/objects", async (ctx) => {
+  const formData = await ctx.req.formData();
+  const file = formData.get("file") as File;
+  // 创建存储空间
+  const createBaseRes = await fetch(storagePluginApi, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      bucket: testBucket,
+      key: file.name,
+      max_content_length: 5242880,
+      metadata: {},
+      min_content_length: 1,
+      post_expires_in_seconds: 3600,
+    }),
+  });
+  if (createBaseRes.ok) {
+    const createBaseData: {
+      object_id: string;
+      storagePluginApi: string;
+      fields: object;
+    } = await createBaseRes.json();
+    const uploadFileForm = new FormData();
+    Object.entries(createBaseData.fields).forEach(([key, value]) => {
+      uploadFileForm.append(key, value);
+    });
+    uploadFileForm.set("file", file);
+    const uploadRes = await fetch(createBaseData.storagePluginApi, {
+      method: "post",
+      body: uploadFileForm,
+    });
+    if (uploadRes.ok) {
+      return ctx.redirect(exampleBaseUrl);
+    }
+    return ctx.text(await uploadRes.text());
+  }
+  return ctx.json(await createBaseRes.json());
```

Comment:
用 early return 来避免嵌套
```suggestion
  if (!createBaseRes.ok) {
    return ctx.json(await createBaseRes.json());
  }

  const createBaseData: {
    object_id: string;
    storagePluginApi: string;
    fields: object;
  } = await createBaseRes.json();

  const uploadFileForm = new FormData();

  Object.entries(createBaseData.fields).forEach(([key, value]) => {
    uploadFileForm.append(key, value);
  });

  uploadFileForm.set("file", file);

  const uploadRes = await fetch(createBaseData.storagePluginApi, {
    method: "post",
    body: uploadFileForm,
  });

  if (uploadRes.ok) {
    return ctx.redirect(exampleBaseUrl);
  } else {
    return ctx.text(await uploadRes.text());
  }
}
```